### PR TITLE
Proper error handling for creating a link without internet

### PR DESCRIPTION
**Pull request by vjeux**

### CODE REVIEW

File: src/scene/data.ts [hunk 1]
```diff
@@ -138,12 +138,18 @@ export async function exportToBackend(
...
+    response = await fetch(BACKEND_POST, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: serializeAsJSON(elements, appState),
+    });
+  } catch (e) {
+    window.alert(i18n.t("alerts.couldNotCreateShareableLink"));
+    return;
+  }
+  const json = await response?.json();
```
Comment (dwelle) [inline src/scene/data.ts]: the optional chain `?.` is either redundant, or if not, then it'll throw on the next line `if(json.id)`

Comment (vjeux) [reply src/scene/data.ts]: typescript doesn’t know that :(

Comment (dwelle) [reply src/scene/data.ts]: Came across my mind, so I tested that specifically on local before commenting :). Works for me.

Comment (dwelle) [reply src/scene/data.ts]: either way, I'll refactor it a bit to be easily readable

Comment (dwelle) [reply src/scene/data.ts]: Check https://github.com/excalidraw/excalidraw/pull/577/commits/b1598e90259ef5c10097aa1251bf6480d8608c4a

this way you don't need to figure out what code paths are possible

Comment (dwelle) [reply src/scene/data.ts]: Also has the added benefit that if anything below the fetch fails, we'll handle it.

**Pull request merged**